include/libtorrent/kademlia/dht_tracker.hpp

```diff
@@ -101,11 +101,18 @@ namespace libtorrent { namespace dht
 			, boost::function<void(item const&, bool)> cb
 			, std::string salt = std::string());
 
+		// for immutable_item.
+		// the callback function will be called when put operation is done.
+		// the bool parameter indicates the put operation success or failed.
 		void put_item(entry data
-			, boost::function<void()> cb);
+			, boost::function<void(int)> cb);
 
+		// for mutable_item.
+		// the data_cb will be called when we get authoritative mutable_item,
+		// the cb is same as put immutable_item.
```

Comment:
My recollection of the semantics of the callback here is that it's called with 'item' containing the latest version of the blob (meaning we have received the authoritative version). The user is expected to potentially modify the item and it will then be put back into the DHT. If the DHT has been updated in between and one or more puts fail with that error, we'll try again.

Is this your understanding too? (sorry, it's been a while since I've been working in this code).